src/Adapter/MSTest.CoreAdapter/Execution/TestMethodInfo.cs

```diff
@@ -171,7 +171,48 @@ public virtual TestResult Invoke(object[] arguments)
 
         internal void SetArguments(object[] arguments)
         {
-            this.arguments = arguments;
+            this.arguments = this.ResolveArguments(arguments);
+        }
+
+        internal object[] ResolveArguments(object[] arguments)
+        {
+            ParameterInfo[] parameterInfos = this.TestMethod.GetParameters();
+            int requiredParameterCount = 0;
+            foreach (var parameter in parameterInfos)
+            {
+                if (!parameter.IsOptional)
+                {
+                    requiredParameterCount++;
+                }
+            }
+
+            if (requiredParameterCount == parameterInfos.Length)
+            {
+                return arguments;
+            }
+
+            if (arguments == null)
```

Comment:
Should we just add this check to SetArguments() function itself and call ResolveArguments() only when `arguments` is not null?

Reply:
moved null check to SetArguments()